packages/api/src/routers/account-recovery/router.ts

```diff
@@ -0,0 +1,265 @@
+  jwt: z.string(),
+})
+
+export const accountRecoveryRouter = createTRPCRouter({
+  getChallenge: publicProcedure.mutation(async () => {
+    const { data: challengeData, error: challengeError } = await insertChallenge()
+
+    if (challengeError || !challengeData) {
+      logger('getChallenge:cant-upsert-challenge')
+      throw new TRPCError({
+        code: 'INTERNAL_SERVER_ERROR',
+        message: formatErr('Cannot generate challenge: Internal server error'),
+        cause: challengeError?.message,
+      })
+    }
+
+    return challengeData as ChallengeResponse
+  }),
+  validateSignature: publicProcedure
+    .input(ValidateSignatureRequestSchema)
+    .mutation(async ({ input, ctx }): Promise<VerifyChallengeResponse> => {
+      const { recoveryType, identifier, signature, challengeId } = input
+
+      const { data: challengeData, error: getChallengeError } = await getChallengeById(challengeId)
+
+      if (getChallengeError || !challengeData) {
+        logger('verifyChallenge:invalid-user-or-challenge')
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: formatErr('Challenge not found'),
+          cause: getChallengeError?.message,
+        })
+      }
+
+      // validate challenge isn't expired
+      await isChallengeExpired(challengeData.id, logger)
+        .then((expired) => {
+          if (expired) {
+            logger('verifyChallenge:challenge-not-found-or-expired')
+            throw new TRPCError({
+              code: 'NOT_FOUND',
+              message: formatErr('Challenge expired'),
+              cause: 'Challenge expired',
+            })
+          }
+        })
+        .catch((error) => {
+          logger('verifyChallenge:challenge-not-found-or-expired')
+          throw new TRPCError({
+            code: 'INTERNAL_SERVER_ERROR',
+            message: formatErr(error),
+            cause: 'Internal server error',
+          })
+        })
+
+      // verify signature with user's identifier
+      const signatureVerified = await verifyRecoverySignature(
+        recoveryType,
+        identifier,
+        signature as `0x${string}`,
+        challengeData
+      )
+
+      // handle unauthorized requests
+      if (!signatureVerified) {
+        logger('api:accountRecoveryRouter:verifyChallenge:signature-failed-verification')
+        throw new TRPCError({
+          code: 'UNAUTHORIZED',
+          message: formatErr(
+            `Unauthorized. Ensure that you are signing with the ${recoveryType} you signed up with`
+          ),
+          cause: 'Signature failed verification',
+        })
+      }
+
+      // identify user from identifier
+      const userId = await getUserIdByIdentifier(recoveryType, identifier)
+      const jwt = mintAuthenticatedJWTToken(userId)
+      const encodedJwt = encodeURIComponent(JSON.stringify([jwt, null, null, null, null, null]))
+
+      logger('api:accountRecoveryRouter:signature-verified')
```

Comment:
let's turn this into a `console.log` and include some context about the request. the debug logger isn't enabled on the server. 

```suggestion
console.log(`account recovered using ${recoveryType} user:${userID} 
```

this way we have minimal monitoring and a short audit trail.